runtime_lib/aircpu/channel.cpp

```diff
@@ -0,0 +1,260 @@
+    air_channel_put<type, 2>(c, s, 0, 0, offset1, offset0,                       \
+                             1, 1, size1, size0, 1, 1,                           \
+                             stride1, stride0);                                  \
+  }
+
+// 1D
+#define mlir_air_channel_get_1d(mangle, type)                                  \
+  void _mlir_ciface_air_channel_get_##mangle(void *c, void *d,                 \
+                                             uint64_t offset0, uint64_t size0, \
+                                             uint64_t stride0)                 \
+  {                                                                            \
+    air_channel_get<type, 1>(c, d, 0, 0, 0, offset0,                           \
+                             1, 1, 1, size0, 1, 1, 1,                          \
+                             stride0);                                         \
+  }
+
+#define mlir_air_channel_put_1d(mangle, type)                                  \
+  void _mlir_ciface_air_channel_put_##mangle(void *c, void *s,                 \
+                                             uint64_t offset0, uint64_t size0, \
+                                             uint64_t stride0)                 \
+  {                                                                            \
+    air_channel_put<type, 1>(c, s, 0, 0, 0, offset0,                           \
+                             1, 1, 1, size0, 1, 1, 1,                          \
+                             stride0);                                         \
+  }
+
+extern "C"
+{
+  // 4D
+  mlir_air_channel_get_4d(M0I64_M0D4I32_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64, int32_t);
+  mlir_air_channel_put_4d(M0I64_M0D4I32_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64, int32_t);
+  mlir_air_channel_get_4d(M0I64_M0D4F32_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64, float);
+  mlir_air_channel_put_4d(M0I64_M0D4F32_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64_I64, float);
+
+  // 3D
+  mlir_air_channel_get_3d(M0I64_M0D3I32_I64_I64_I64_I64_I64_I64_I64_I64_I64, int32_t);
+  mlir_air_channel_put_3d(M0I64_M0D3I32_I64_I64_I64_I64_I64_I64_I64_I64_I64, int32_t);
+  mlir_air_channel_get_3d(M0I64_M0D3F32_I64_I64_I64_I64_I64_I64_I64_I64_I64, float);
+  mlir_air_channel_put_3d(M0I64_M0D3F32_I64_I64_I64_I64_I64_I64_I64_I64_I64, float);
+
+  // 2D
+  mlir_air_channel_get_2d(M0I64_M0D2I32_I64_I64_I64_I64_I64_I64, int32_t);
+  mlir_air_channel_put_2d(M0I64_M0D2I32_I64_I64_I64_I64_I64_I64, int32_t);
+  mlir_air_channel_get_2d(M0I64_M0D2F32_I64_I64_I64_I64_I64_I64, float);
+  mlir_air_channel_put_2d(M0I64_M0D2F32_I64_I64_I64_I64_I64_I64, float);
+
+  // 1D
+  mlir_air_channel_get_1d(M0I64_M0D1I32_I64_I64_I64, int32_t);
+  mlir_air_channel_put_1d(M0I64_M0D1I32_I64_I64_I64, int32_t);
+  mlir_air_channel_get_1d(M0I64_M0D1F32_I64_I64_I64, float);
+  mlir_air_channel_put_1d(M0I64_M0D1F32_I64_I64_I64, float);
```

Comment:
Just curious: would it be possible to use just explicit template instantiation instead of macros? I have the feeling it would synthesize automatically the right mangling according to the ABI. Or is it just some ACDC-private kind of mangling?